ls1intum/Artemis
# `Development`: Refactor result detail component

### Checklist
#### General


#### Client

### Motivation and Context
Follow-up to #4835 where I noticed the pretty long methods in this component.

### Description
Makes the code more readable by splitting up methods. Adds additional unit tests for previously uncovered code elements.

Additionally lets the component handle cases where no exercise is given. I went through all usages of the component to pass an exercise wherever it is known so that the chart can still be shown in nearly all cases.

### Steps for Testing

Prerequisites:
- 1 Instructor
- Some programming exercise either already graded or not, optionally with static code analysis enabled.

1. Log in to Artemis
2. All feedback types (test case feedback, SCA feedback, manual feedback with/without grading instruction usage) should still be shown as usual in the dialogue when clicking on a result string (e.g., ‘12 of 12 passed, 100%’). No errors about a missing exercise should be printed in the console log.
   Those result strings are shown for example
   - in the online editor of programming exercises,
   - on the ‘Submissions’ page of an exercise accessible to instructors,
   - in the student’s view when looking at all exercises for a course,
   - in the student’s view when looking at an exercise they submitted to.

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage


| Class/File | Branch | Line |
|------------|-------:|-----:|
| `result-detail.component.ts` | 87.7% | 94.4% |
